project/src/main/java/com/google/sps/servlets/EventCreationServlet.java

```diff
@@ -0,0 +1,68 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.cloud.Date;
+import com.google.gson.Gson;
+import com.google.sps.data.Event;
+import com.google.sps.data.User;
+import com.google.sps.utilities.DatabaseWrapper;
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/create-event")
+public class EventCreationServlet extends HttpServlet {
+  private static final DatabaseWrapper dbWrapper = new DatabaseWrapper("step-188-instance", "event-organizer-db");
+
+  /** Returns event details from database */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String eventId = request.getParameter("eventId");
+    Optional<Event> eventOptional = dbWrapper.getEventById(eventId);
+    Event event;
+
+    // If event DNE, should never happen because backend passes Ids
+    if (!eventOptional.isPresent()) {
+        event = new Event.Builder("ERROR: This event does not exist", "", new HashSet<String>(), "", null, null).build();
```

Comment:
This is kind of against the Builder rule since you are building a Event without lots of required fields.

Can you figure out a better solution for this? Maybe throw an Exception? return an Optional?

Reply:
yes, that makes sense. will send 404 that gets processed in frontend.